redisinsight/api/src/modules/cloud/database/cloud-database.analytics.ts

```diff
@@ -13,7 +16,27 @@ export class CloudDatabaseAnalytics extends TelemetryBaseService {
     this.sendEvent(TelemetryEvents.CloudFreeDatabaseCreated, eventData);
   }
 
-  sendCloudFreeDatabaseFailed(exception: HttpException, eventData: object = {}) {
-    this.sendFailedEvent(TelemetryEvents.CloudFreeDatabaseFailed, exception, eventData);
+  async sendCloudFreeDatabaseFailed(
+    exception: HttpException,
+    eventData: object = {},
+    cloudSubscriptionCapiService: CloudSubscriptionCapiService,
+    data: { planId: number, capiCredentials: CloudCapiAuthDto },
+  ) {
+    const fixedPlans = await cloudSubscriptionCapiService.getSubscriptionsPlans(
```

Comment:
I'm not with DI inside analytics. We avoided such cases till this PR...
From my point of view we should get all required data before sendCloudFreeDatabaseFailed invocation or do not send additional fields. 

If it is urgent I'll approve this with assumption that in the nearest future it will be implemented in according suggestions above BUT you have to wrap with `try {} catch {}` this function since we might crash entire api server with unhandled promise rejection.